api_tests/harness.js

```diff
@@ -16,202 +20,267 @@ const project_root = execSync("git rev-parse --show-toplevel", {
 }).trim();
 global.harness.project_root = project_root;
 
-const test_dir = process.env.TEST_DIR;
-if (!test_dir) {
-    throw new Error("env var $TEST_DIR env variable must be set");
-}
-
-const log_dir = test_dir + "/log";
-global.harness.log_dir = log_dir;
-
-const log4js = require("log4js");
-log4js.configure({
-    appenders: {
-        test_suite: {
-            type: "file",
-            filename: log_dir + "/test-suite.log",
-        },
-    },
-    categories: { default: { appenders: ["test_suite"], level: "ALL" } },
-});
-
 const docker_cwd = project_root + "/api_tests/regtest";
 const services_cwd = project_root + "/api_tests/";
-process.chdir(services_cwd);
+global.harness.test_root = services_cwd;
 
 const docker_compose_options = {
     cwd: docker_cwd,
     encoding: "utf-8",
 };
 
-const config = Toml.parse(fs.readFileSync(test_dir + "/config.toml", "utf8"));
-global.harness.config = config;
-
 const ledgers_config = Toml.parse(
     fs.readFileSync(project_root + "/api_tests/regtest/ledgers.toml", "utf8")
 );
 global.harness.ledgers_config = ledgers_config;
 
-let ledger_up_time = 0;
-let ledger_down_time = 0;
-{
-    if (config.ledgers) {
-        config.ledgers.forEach(function(ledger) {
-            const config_ledger = ledgers_config[ledger];
-
-            const up_time = config_ledger.up_time;
-            if (up_time && up_time > ledger_up_time) {
-                ledger_up_time = up_time;
-            }
-
-            const down_time = parseInt(config_ledger.down_time);
-            if (down_time && down_time > ledger_down_time) {
-                ledger_down_time = down_time;
-            }
-        });
-    }
-}
-
 // To be done once all global variables are set
 const util = require("./lib/util.js");
+const log_dir = project_root + "/api_tests/log";
 
-// *********************************************** //
-// Clean-up docker containers and processes helper //
-// *********************************************** //
-
-function cleanUp(subprocesses) {
-    subprocesses.forEach(function(subprocess) {
-        subprocess.kill();
-    });
-    execSync("docker-compose rm -sfv", docker_compose_options);
+if (!fs.existsSync(log_dir)) {
+    fs.mkdirSync(log_dir);
 }
 
-process.once("SIGINT", function() {
-    cleanUp();
-});
-
 // ********************** //
 // Start services helpers //
 // ********************** //
+class LedgerRunner {
+    constructor(ledgers_config) {
+        this.running_ledgers = {};
+        this.ledgers_config = ledgers_config;
+        this.block_timers = {};
+    }
+
+    async ensure_ledgers_running(ledgers) {
```

Comment:
```suggestion
    async ensureLedgersRunning(ledgers) {
```
Let's use JS appropriate naming

Reply:
Are you asking me to change everything from snake case to camel case? Or just not to make things worse?

If we're doing this we should probably use a linter to enforce this.

Reply:
use camelCase for new variables/functions.
No need to refactor everything.
Good idea for the linter, we can do that later.